### ISSUE TITLE
Feature request:  FSDP for TPUs

### ISSUE THREAD

**Issue by OhadRubin** [huggingface/accelerate#422]
A recent contribution to the pytorch_xla repo allows using FSDP in PyTorch XLA for sharding Module parameters across data-parallel workers. https://github.com/pytorch/xla/pull/3431
What will be needed to add it to accelerate?

**Comment by muellerzr** [comment]
Once the next release of PyTorch XLA is out, we'll start taking a look at this

**Comment by Vatshank** [comment]
Hey @muellerzr, is there ongoing work for adding [XLA support](https://github.com/pytorch/xla/pull/3431) to FSDP? We, on the AWS SageMaker training compiler side, have started looking into XLA-FSDP and might be able to contribute to adding such support to accelerate.

**Comment by muellerzr** [comment]
@Vatshank not yet! It's the next thing on my list to get to after TPU pod support, so would love the help if you guys can! 🙏

**Comment by Vatshank** [comment]
Okay cool! Although our focus is on GPUs, I am sure there will be significant overlap in the code for adding support for either device type.

What do you think would be a good way to discuss some of these details? If you guys have a shared Slack group for development, for instance. Also happy to bug you on GitHub, if that's preferred :)

**Comment by muellerzr** [comment]
@Vatshank this gh issue should be fine!

**Comment by JackCaoG** [comment]
@AlexWertheim With your recent pr can we call this request done?

**Comment by AlexWertheim** [comment]
> @AlexWertheim With your recent pr can we call this request done?

Yeah, I think so. For reference, the PR in question can be seen [here](https://github.com/huggingface/transformers/pull/21406). @muellerzr can say better than I can whether this fulfills all requirements where accelerate is concerned.
